libraries/DateTime/test/dateTime.js

```diff
@@ -0,0 +1,344 @@
+const Ae = require('@aeternity/aepp-sdk').Universal
+
+const config = {
+  host: 'http://localhost:3001/',
+  internalHost: 'http://localhost:3001/internal/',
+  gas: 200000,
+  ttl: 55,
+}
+let contractSource = utils.readFileRelative('./contracts/DateTime.aes', 'utf-8')
+async function deployContract(owner) {
+  const compiledContract = await owner.contractCompile(contractSource, {
+    gas: config.gas,
+  })
+  const deployPromise = compiledContract.deploy({
+    options: {
+      ttl: config.ttl,
+    },
+  })
+  return deployPromise
+}
+
+async function callContract(contract, contract_name, timestamp) {
+  const result = await contract.call(contract_name, {
+    args: `(${timestamp})`,
+    options: { ttl: config.ttl },
+    abi: 'sophia',
+  })
+  return result
+}
+
+async function processArray(array, function_name) {
+  let data = []
+  for (const item of array) {
+    data.push(await function_name(item, item.timestamp))
+  }
+  return data.map(el => el.value)
+}
+
+describe('Test DateTime', () => {
+  let owner = null
+
+  before(async () => {
+    const ownerKeyPair = wallets[0]
+    owner = await Ae({
+      url: config.host,
+      internalUrl: config.internalHost,
+      keypair: ownerKeyPair,
+      nativeMode: true,
+      networkId: 'ae_devnet',
+    })
+  })
+
+  describe('Test convert timestamp', () => {
+    it('parse_timestamp function', async () => {
+      const year = 2016
+      const month = 3
+      const day = 7
+      const hour = 8
+      const minute = 58
+      const second = 47
+      const weekday = 1
+      const timestamp = 1457341127
+
+      const contract = await deployContract(owner)
+      const callWithdraw = await callContract(
+        contract,
+        'parse_timestamp',
+        timestamp
+      )
+      const decodedWithdraw = await callWithdraw
+        .decode('(int, int, int, int, int, int, int, int)')
+        .then(item => item.value.map(el => el.value))
```

Comment:
do not use 'async/await' and 'then' together....

Reply:
const result = await callWithdraw
        .decode('(int, int, int, int, int, int, int, int)');
        
      const decodedWithdraw =  result.value.map(item => item.value);